app/services/collect_inventories_service.rb

```diff
@@ -0,0 +1,44 @@
+class CollectInventoriesService
+  attr_reader :inventory_ids
+
+  def initialize(options)
+    @options = options.deep_symbolize_keys
+    @service_offering_id = @options[:service_offering_id]
+    @task_id = @options[:task_id]
+    @inventory_ids = []
+  end
+
+  def process
+    Task.update(@task_id, :state => "running", :status => "ok")
+
+    visited = []
+    collect_inventory(@service_offering_id, visited, @inventory_ids)
+
+    Task.update(@task_id, :state => "completed", :status => "ok", :output => {:applied_inventories => @inventory_ids})
```

Comment:
@hsong-rh We should be returning tags here and not ids. Catalog is going to get these ids and then it will make another call to get the tags. So we can return the array of tag objects

```
inventory_ids.each.collect { |id| ServiceInventory.find(id).tags }.flatten
```

Reply:
@hsong-rh 
This is where catalog is going to use it
https://github.com/RedHatInsights/catalog-api/blob/f45a57e950de1fe9a48cf9dec0d599e17c880620/lib/tags/topology/remote_inventory.rb#L33
We can save the extra set of calls by just returning the tags